engine/src/main/java/com/datatorrent/stram/ClusterModeImpl.java

```
@@ -0,0 +1,100 @@
+
+import com.datatorrent.api.Attribute;
+import com.datatorrent.api.ClusterMode;
+import com.datatorrent.api.StreamingApplication;
+import com.datatorrent.stram.client.StramAppLauncher;
+import com.datatorrent.stram.plan.logical.LogicalPlan;
+import com.datatorrent.stram.plan.logical.LogicalPlanConfiguration;
+import com.datatorrent.stram.util.StreamingAppFactory;
+
+/**
+ *
+ */
+public class ClusterModeImpl extends ClusterMode<ClusterAppHandle>
+{
+
+  private static final Map<Attribute<?>, String> propMapping = new HashMap<>();
+
+  static {
+    propMapping.put(ClusterMode.LIB_JARS, StramAppLauncher.LIBJARS_CONF_KEY_NAME);
+    propMapping.put(ClusterMode.ORIGINAL_APP_ID, StramAppLauncher.ORIGINAL_APP_ID);
+    propMapping.put(ClusterMode.QUEUE_NAME, StramAppLauncher.QUEUE_NAME);
+  }
+
+  public ClusterAppHandle launchApp(final StreamingApplication app, Configuration conf, Attribute.AttributeMap launchAttributes) throws LaunchException
+  {
+    if (launchAttributes != null) {
+      for (Map.Entry<Attribute<?>, Object> entry : launchAttributes.entrySet()) {
+        String property = propMapping.get(entry.getKey());
+        if (property != null) {
+          setConfiguration(conf, property, entry.getValue());
+        }
+      }
+    }
+    try {
+      String name = app.getClass().getName();
+      StramAppLauncher appLauncher = new StramAppLauncher(name, conf);
+      appLauncher.loadDependencies();
+      StreamingAppFactory appFactory = new StreamingAppFactory(name, app.getClass())
+      {
+        @Override
+        public LogicalPlan createApp(LogicalPlanConfiguration planConfig)
+        {
+          return super.createApp(app, planConfig);
+        }
+      };
+      ApplicationId appId = appLauncher.launchApp(appFactory);
+      return new ClusterAppHandle(appId);
+    } catch (Exception ex) {
+      throw new LaunchException(ex);
+    }
+  }
+
+  @Override
+  public void shutdownApp(ClusterAppHandle app)
+  {
+    // TODO
```

if something isn't implemented, it should throw UnsupportedOperation